Add-ons/UmbracoForms/Developer/Rendering-Scripts/index.md

```diff
@@ -5,26 +5,59 @@ versionTo: 10.0.0
 
-In many cases, you might prefer rendering your scripts at the bottom of the page, e.g. before the closing `</body>` tag, as this generally improves site performance.
+```csharp
+@model Umbraco.Forms.Web.Models.FormViewModel
 
-In order to render your scripts where you want, you need to add the following snippet to your template. Make sure you add it below your scripts, right before the closing `</body>` tag:
+@using Newtonsoft.Json
+@using Umbraco.Forms.Core
+@using Umbraco.Forms.Web
 
-```csharp
-@using Umbraco.Forms.Web.Extensions;
+@{
+    Html.AddFormThemeScriptFile("~/App_Plugins/UmbracoForms/Assets/themes/default/umbracoforms.js");
+}
 
-@if (TempData["UmbracoForms"] != null)
+<div class="umbraco-forms-form-config"
+     style="display: none"
+     data-id="@Model.FormClientId"
+     data-serialized-page-button-conditions="@JsonConvert.SerializeObject(Model.PageButtonConditions, FormsJsonSerializerSettings.Default)"
+     data-serialized-fieldset-conditions="@JsonConvert.SerializeObject(Model.FieldsetConditions, FormsJsonSerializerSettings.Default)"
+     data-serialized-field-conditions="@JsonConvert.SerializeObject(Model.FieldConditions, FormsJsonSerializerSettings.Default)"
+     data-serialized-fields-not-displayed="@JsonConvert.SerializeObject(Model.GetFieldsNotDisplayed(), FormsJsonSerializerSettings.Default)"
+     data-trigger-conditions-check-on="@Model.TriggerConditionsCheckOn"
+     data-form-element-html-id-prefix="@Model.FormElementHtmlIdPrefix"></div>
+
+@* Only render out scripts on the page if the form has not been submitted yet *@
+@if (Model.SubmitHandled == false)
 {
-    foreach (var form in TempData.Get<List<Guid>>("UmbracoForms"))
+    @*
+        If the current page of the form has any Partial view files attached to render
+        Likely used by events and third parties adding tracking or other 3rd party functionality to a form
+    *@
+    if (Model.CurrentPage.PartialViewFiles.Any())
     {
-        @await Component.InvokeAsync("RenderFormScripts", new {formId = form, theme = "bootstrap3-horizontal"})
+        foreach (var partial in Model.CurrentPage.PartialViewFiles)
+        {
+            @await Html.PartialAsync(partial.Value)
+        }
     }
+
+    @* Render references to javascript files needed by fields on the current page*@
+    @Html.RenderFormsScripts(Url, Model, Model.JavaScriptTagAttributes)
+    @Html.RenderFormsStylesheets(Url, Model)
 }
+
 ```
 
+:::note
+If you are upgrading from an older version of Umbraco Forms and using a variant of `Scripts.cshtml` within your own project you may be missing some required and fundamental parts of this file. As an example, the `<div class="umbraco-forms-form-config" />` element introduced the `data-form-element-html-id-prefix` attribute in Umbraco Forms 10.2. Failing to update this within your own version of this file may lead to a loss of functionality and/or JavaScript errors.
```

Comment:
This is fair point, but very specific about a particular version and issue - and really the point is general.  If you're using a custom theme, you will want to review when the default one is updated to make sure you apply to your custom theme what changes may be necessary

Rather than adding this, probably something to that effect can be usefully added [here](https://our.umbraco.com/documentation/Add-ons/UmbracoForms/Developer/Themes/#creating-a-theme).